tests/rptest/services/redpanda.py

```diff
@@ -1940,20 +1940,25 @@ def save_executable(self):
             self._saved_executable = True
             self._context.log_collect['executable', self] = True
 
-    def search_log(self, pattern):
-        """
-        Test helper for grepping the redpanda log
+    def search_log(self, pattern: str, nodes: list[ClusterNode] = None):
+        # Test helper for grepping the redpanda log
 
-        :return:  true if any instances of `pattern` found
-        """
-        for node in self.nodes:
+        # :param pattern: the string to search for
+        # :param nodes: a list of nodes to run grep on
+
+        # :return:  true if `pattern` is found in all nodes
+        matches = 0
+
+        if nodes is None:
+            nodes = self.nodes
+
+        for node in nodes:
             for line in node.account.ssh_capture(
                     f"grep \"{pattern}\" {RedpandaService.STDOUT_STDERR_CAPTURE} || true"
             ):
                 # We got a match
                 self.logger.debug(
                     f"Found {pattern} on node {node.name}: {line}")
-                return True
+                matches += 1
 
-        # Fall through, no matches
-        return False
+        return matches > 0
```

Comment:
This seems in incongruous with: "# :return:  true if `pattern` is found in all nodes", the condition here is any node.

Does this slow down the test suite by searching all nodes, even if the old behaviour of first match is desired?

Reply:
> This seems in incongruous with: "# :return: true if `pattern` is found in all nodes", the condition here is any node.

Good catch. Changed.

> Does this slow down the test suite by searching all nodes

Yes it does slowdown the test suite but I figured that was an acceptable trade-off for a stronger check.

> , even if the old behaviour of first match is desired?

We don't have many ducktape tests that use search_log (See cluster_config_test.py and liscense_upgrade_test.py). The tests that do call search_log do so in a context of checking conditions on **all brokers**. So I figured the old behavior of first match was no longer desired. I'll check this before my next push.